### PR TITLE
Speech Recognition - initial commit

### PR DESCRIPTION
**Pull request by kubawolanin**
Signed-off-by: Kuba Wolanin <hi@kubawolanin.com>

A few remarks:
* For locale I'm utilizing `/rest/services/org.eclipse.smarthome.core.localeprovider/config` 
* I'm not relying on `VoiceCommand` item (as discussed in #80) - I'm passing data to `/voice/interpreters` instead.

### CODE REVIEW

**Comment by ghys** [comment]
I need to test it more but it does recognizes my voice! :smile:
Chrome on my Android tablet asks for permission to use the microphone every time, surely I can add an exception, I just don't know how?
I'll send my remarks later, stay tuned.

**Comment by kubawolanin** [comment]
> Chrome on my Android tablet asks for permission to use the microphone every time, surely I can add an exception, I just don't know how?

I think you need to go to Settings / Site settings and [set the priviliges for a given url or all websites](https://support.google.com/chrome/answer/114662?co=GENIE.Platform%3DAndroid&hl=en&oco=1).

**Comment by ghys** [comment]
image
The overlay seems misaligned, is this normal?

File: web/app/dashboard/dashboard.view.controller.js [hunk 1]
```diff
@@ -63,4 +64,64 @@
...
+            }
+        };
+
+        speech.onend = function(e) {
+            console.log('Stopped listening');
+            vm.speechOutput = '';
+            vm.isListening = false;
+        }
+    }
   }
```
Comment (ghys) [inline web/app/dashboard/dashboard.view.controller.js]: This (or most of it) should probably go to the `speech.service.js` file and the controller should just call it, don't you think?

File: web/app/services/openhab.service.js [hunk 2]
```diff
@@ -165,15 +205,15 @@
...
+                url                        : '/rest/items',
+                contentType                : 'application/json',
+                logLevel                   : 'debug',
+                transport                  : 'websocket',
+                fallbackTransport          : 'long-polling',
+                attachHeadersAsQueryString : true,
+                reconnectInterval          : 5000,
+                enableXDR                  : true,
+                timeout                    : 60000
             };
```
Comment (ghys) [inline web/app/services/openhab.service.js]: not a fan of this formatting tbh ;) (but I don't care that much)

Comment (kubawolanin) [inline web/app/services/openhab.service.js]: Much cleaner for me to read ;) I'm using AlignTab for Sublime.

File: web/app/dashboard/dashboard.view.controller.js (hunk 1 above)
Comment (kubawolanin) [inline web/app/dashboard/dashboard.view.controller.js]: Yes, I even tried that but failed. Maybe you'd have a suggestion how to handle passing values to `vm.speechOutput` and `vm.isListening`?
And then there's this `OHService.sendVoice(vm.speechOutput);` which inside `speech.service.js` would result in circular reference (I guess).

**Comment by kubawolanin** [comment]
Hi Yannick, thanks for the review!
I'll take a look at the styling - it should be displayed on the bottom of the screen actually.

**Comment by ghys** [comment]
Cool! I made the refactoring in the above commit, I'll let you take care of the overlay though.

File: web/app/dashboard/dashboard.view.controller.js (hunk 1 above)
Comment (ghys) [inline web/app/dashboard/dashboard.view.controller.js]: Did it below :)

File: web/assets/styles/dashboard.scss [hunk 3]
```diff
@@ -34,6 +34,21 @@
...
+  font-weight: 300;
+  background: linear-gradient(to bottom, rgba(0,0,0,0) 0%,rgba(0,0,0,1) 100%);
+  width: 100%;
+  bottom: 0;
+  z-index: 400;
+  padding: 15px 0;
+  margin-left: -15px;
+}
+
 .controls {
```
Comment (ghys) [inline web/assets/styles/dashboard.scss]: Display issues as discussed above...

Comment (kubawolanin) [inline web/assets/styles/dashboard.scss]: Done! Turns out I forgot to rebuild SCSS files :)

**Pull request merged**